python/test/backend/extension_backend.c

```diff
@@ -0,0 +1,42 @@
+#include <Python.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+static PyObject *getDeviceProperties(PyObject *self, PyObject *args) {
+  // create a struct to hold device properties
+  return Py_BuildValue("{s:i, s:i, s:i, s:i, s:i}", "max_shared_mem", 1024,
```

Comment:
where do these values come from?

Reply:
It is a dummy value and is only for test purposes. In a real case, it should come from the hardware info.